### ISSUE TITLE
v6.0.1 -> The package may have incorrect main/module/exports specified

### ISSUE THREAD

**Issue by jochensengier** [ModyQyW/vite-plugin-stylelint#62]
### Describe the bug

There's an issue with the new v6.0.1.
Currently rolling back to v6.0.0

```
> vite build --config vite.front.config.mjs

✘ [ERROR] Failed to resolve entry for package "vite-plugin-stylelint". The package may have incorrect main/module/exports specified in its package.json. [plugin externalize-deps]

    node_modules/esbuild/lib/main.js:1213:27:
      1213 │         let result = await callback({
           ╵                            ^

  This error came from the "onResolve" callback registered here:

    node_modules/esbuild/lib/main.js:1137:20:
      1137 │       let promise = setup({
           ╵                     ^

  The plugin "externalize-deps" was triggered by this import

    vite.front.config.mjs:6:22:
      6 │ import stylelint from 'vite-plugin-stylelint'
        ╵                       ~~~~~~~~~~~~~~~~~~~~~~~
```

### Reproduction

> vite build --config vite.front.config.mjs

### System Info

```Shell
System:
    OS: macOS 15.5
    CPU: (10) arm64 Apple M2 Pro
    Memory: 1.05 GB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.16.0 - ~/.nvm/versions/node/v22.16.0/bin/node
    npm: 11.4.2 - ~/.nvm/versions/node/v22.16.0/bin/npm
  Browsers:
    Chrome: 137.0.7151.122
    Safari: 18.5
```

### Used Package Manager

npm

### Validations

- [x] Follow our [Code of Conduct](https://github.com/uni-helper/.github/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://github.com/antfu/contribute) and [Notes from a tired maintainer](https://github.com/pi0/tired-maintainer).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [x] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

**Comment by Anoesj** [comment]
Running into a similar issue where `@nuxtjs/stylelint-module` causes the following error:
```
[nuxi 1:39:16 PM]  ERROR  Cannot start nuxt:  Cannot find module '/path/obfuscated/node_modules/.pnpm/@nuxtjs+stylelint-module@5.2.1_magicast@0.3.5_postcss@8.5.6_rollup@4.44.1_stylelint@16._af286cfe42a10976a169cb3f220e5616/node_modules/vite-plugin-stylelint/dist/index.mjs' imported from /path/obfuscated/node_modules/.pnpm/@nuxtjs+stylelint-module@5.2.1_magicast@0.3.5_postcss@8.5.6_rollup@4.44.1_stylelint@16._af286cfe42a10976a169cb3f220e5616/node_modules/@nuxtjs/stylelint-module/dist/module.mjs
Did you mean to import "vite-plugin-stylelint/dist/index.cjs"?
```

Furthermore, shouldn't vite 7 support have been released as a new major or minor version? This was released as a patch update, but I'm not sure I agree with that.

**Comment by brybrant** [comment]
The package.json points to `index.mjs` - but no file with that extension exists in the package. Instead, the build outputs `index.js`

https://github.com/ModyQyW/vite-plugin-stylelint/blob/main/packages/core/package.json#L28

**Comment by ModyQyW** [comment]
@brybrant Thank you! It is my bad. The fix is launched in v6.0.2.

**Comment by ModyQyW** [comment]
@Anoesj Originally, this was just a small peerDependencies update, containing no breaking changes or new features, so I released a patch.
